Kube-proxy/iptables: unit tests fails for count>1

### What happened?

When unit tests in `kubernetes/pkg/proxy/iptables` are invoked with `--count` > 1, it fails

### What did you expect to happen?

Unit tests should work for any `count`. This is important for troubleshooting flaking tests.

### How can we reproduce it (as minimally and precisely as possible)?

```
cd kubernetes/pkg/proxy/iptables
go test -count=2
I1108 09:09:52.713245  154009 proxier.go:809] "Sync failed" retryingTime="0s"
--- FAIL: TestSyncProxyRulesRepeated (0.00s)
    proxier_test.go:6685: Already did a partial resync? Something failed earlier!
FAIL
exit status 1
FAIL    k8s.io/kubernetes/pkg/proxy/iptables    0.432s
# count=1 works
go test 
ok      k8s.io/kubernetes/pkg/proxy/iptables    0.198s
```



### Anything else we need to know?

Discovered when I tried to add parallel execution as described in https://github.com/kubernetes/kubernetes/pull/121791

### Kubernetes version

master

### Cloud provider

N/A

### OS version

N/A

### Install tools

N/A

### Container runtime (CRI) and version (if applicable)

N/A

### Related plugins (CNI, CSI, ...) and versions (if applicable)

N/A

--- a/pkg/proxy/iptables/proxier_test.go
+++ b/pkg/proxy/iptables/proxier_test.go
@@ -40,6 +40,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/component-base/metrics/legacyregistry"
 	"k8s.io/component-base/metrics/testutil"
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/features"
@@ -6150,6 +6151,7 @@ func TestSyncProxyRulesRepeated(t *testing.T) {
 	ipt := iptablestest.NewFake()
 	fp := NewFakeProxier(ipt)
 	metrics.RegisterMetrics()
+	defer legacyregistry.Reset()
 
 	// Create initial state
 	var svc2 *v1.Service
@@ -6677,11 +6679,11 @@ func TestSyncProxyRulesRepeated(t *testing.T) {
 	if fp.needFullSync {
 		t.Fatalf("Proxier unexpectedly already needs a full sync?")
 	}
-	prFailures, err := testutil.GetCounterMetricValue(metrics.IptablesPartialRestoreFailuresTotal)
+	partialRestoreFailures, err := testutil.GetCounterMetricValue(metrics.IptablesPartialRestoreFailuresTotal)
 	if err != nil {
 		t.Fatalf("Could not get partial restore failures metric: %v", err)
 	}
-	if prFailures != 0.0 {
+	if partialRestoreFailures != 0.0 {
 		t.Errorf("Already did a partial resync? Something failed earlier!")
 	}
 
@@ -6711,12 +6713,12 @@ func TestSyncProxyRulesRepeated(t *testing.T) {
 	if !fp.needFullSync {
 		t.Errorf("Proxier did not fail on previous partial resync?")
 	}
-	updatedPRFailures, err := testutil.GetCounterMetricValue(metrics.IptablesPartialRestoreFailuresTotal)
+	updatedPartialRestoreFailures, err := testutil.GetCounterMetricValue(metrics.IptablesPartialRestoreFailuresTotal)
 	if err != nil {
 		t.Errorf("Could not get partial restore failures metric: %v", err)
 	}
-	if updatedPRFailures != prFailures+1.0 {
-		t.Errorf("Partial restore failures metric was not incremented after failed partial resync (expected %.02f, got %.02f)", prFailures+1.0, updatedPRFailures)
+	if updatedPartialRestoreFailures != partialRestoreFailures+1.0 {
+		t.Errorf("Partial restore failures metric was not incremented after failed partial resync (expected %.02f, got %.02f)", partialRestoreFailures+1.0, updatedPartialRestoreFailures)
 	}
 
 	// On retry we should do a full resync, which should succeed (and delete svc4)

